aggregator/rpc_server/server.go

```diff
@@ -0,0 +1,170 @@
+package rpc_server
+
+import (
+	"errors"
+	"fmt"
+	"github.com/NethermindEth/near-sffl/core"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"net/rpc"
+	"strings"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	eigentypes "github.com/Layr-Labs/eigensdk-go/types"
+
+	"github.com/NethermindEth/near-sffl/aggregator"
+	"github.com/NethermindEth/near-sffl/core/types/messages"
+)
+
+var (
+	TaskNotFoundError400                     = errors.New("400. Task not found")
+	OperatorNotPartOfTaskQuorum400           = errors.New("400. Operator not part of quorum")
+	TaskResponseDigestNotFoundError500       = errors.New("500. Failed to get task response digest")
+	MessageDigestNotFoundError500            = errors.New("500. Failed to get message digest")
+	OperatorSetUpdateBlockNotFoundError500   = errors.New("500. Failed to get operator set update block")
+	UnknownErrorWhileVerifyingSignature400   = errors.New("400. Failed to verify signature")
+	SignatureVerificationFailed400           = errors.New("400. Signature verification failed")
+	CallToGetCheckSignaturesIndicesFailed500 = errors.New("500. Failed to get check signatures indices")
+	UnknownError400                          = errors.New("400. Unknown error")
+
+	errorsMap = map[error]error{
+		aggregator.DigestError:                    MessageDigestNotFoundError500,
+		aggregator.TaskResponseDigestError:        TaskResponseDigestNotFoundError500,
+		aggregator.GetOperatorSetUpdateBlockError: OperatorSetUpdateBlockNotFoundError500,
+	}
+)
+
+type RpcServer struct {
+	serverIpPortAddr string
+	app              aggregator.RpcAggregatorer
+
+	logger   logging.Logger
+	listener EventListener
+}
+
+var _ core.Metricable = (*RpcServer)(nil)
+
+func NewRpcServer(serverIpPortAddr string, app aggregator.RpcAggregatorer, logger logging.Logger) *RpcServer {
+	return &RpcServer{
+		serverIpPortAddr: serverIpPortAddr,
+		app:              app,
+		logger:           logger,
+		listener:         &SelectiveRpcListener{},
+	}
+}
+
+func (s *RpcServer) EnableMetrics(registry *prometheus.Registry) error {
+	listener, err := MakeRpcServerMetrics(registry)
+	if err != nil {
+		return err
+	}
+
+	s.listener = listener
+	return nil
+}
+
+func (s *RpcServer) Start() error {
+	s.logger.Info("Starting aggregator rpc server.")
+
+	err := rpc.Register(s)
```

Comment:
Checking this PR deeply later, but wanted to highlight something I noticed.

Here we are changing the name used for the RPC calls (can be seen in `rpc_client.go`). I'd still prefer it to be registered as `Aggregator`, as it's the aggregator RPC server indeed, but either way - keeping it like this is a **breaking change**. I think it wouldn't be if we kept the name.
That applies to all PRs from everyone: whenever what you're doing is a breaking change, make this clear somehow so it's properly handled, especially in the PR description. Reviewers should also be aware of this whenever reviewing, as it may pass unnoticed by the author. @taco-paco @emlautarom1 @mikhailUshakoff

Reply:
Agree